discovery_engine/lib/src/domain/search_manager.dart

```diff
@@ -80,7 +80,9 @@ class SearchManager {
     );
     await _engineStateRepo.save(await _engine.serialize());
 
-    await _docRepo.updateMany(searchDocs.map((e) => e.document));
+    final documents = searchDocs.map((e) => e.document..isSearched = true);
```

Comment:
Interesting use of the cascade operator. I personally would split this into multiple lines, for readability sake, but it's not a must.